packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -1115,8 +1115,8 @@ function detachFiber(current: Fiber) {
   current.pendingProps = null;
   current.memoizedProps = null;
   current.stateNode = null;
-  if (alternate !== null) {
-    detachFiber(alternate);
+  if (detatchAlternate && alternate !== null) {
+    detachFiber(alternate, false);
```

Comment:
Instead of boolean, I would remove these lines and call `detachFiber` on each fiber separately.

```js
detachFiber(fiber);
if (fiber.alternate !== null) {
  detachFiber(fiber.alternate);
}
```